tests/camera.cpp

```diff
@@ -0,0 +1,168 @@
+/** @file
+ *  Program entry point.
+ *
+ *  @copyright 2006  Palm Studios
+ *
+ *  @author Jesse Buhagiar
+ */
+
+#include "SH3/arc/mft.hpp"
+#include "SH3/system/config.hpp"
+#include "SH3/system/exit_code.hpp"
+#include "SH3/system/log.hpp"
+#include "SH3/system/window.hpp"
+#include "SH3/system/glprogram.hpp"
+#include "SH3/system/glbuffer.hpp"
+#include "SH3/system/glvertarray.hpp"
+#include "SDL2/SDL.h"
+#include "SH3/camera/camera.hpp"
+#include <cstdio>
+
+static GLfloat g_vertex_buffer_data[] = {
+   -1.0f, -1.0f, 0.0f,
+   1.0f, -1.0f, 0.0f,
+   0.0f,  1.0f, 0.0f,
+};
+
+static GLfloat glCol[] = {
+    1.0f, 0.0f, 0.0f,
+    0.0f, 1.0f, 0.0f,
+    0.0f, 0.0f, 1.0f,
+};
+
+/**
+ *  VAO Attributes for a colorful triangle.
+ */
+struct TriangleAttributes final
+{
+private:
+    using Target = sh3_gl::buffer_object::Target;
+
+    TriangleAttributes() = delete;
+
+public:
+    enum class Slot
+    {
+        VERTEX,
+        COLOR,
+        MAX
+    };
+
+    static constexpr sh3_gl::vao_target_array<Slot> Targets =
+    {
```

Comment:
I'm still getting the `-Wmissing-braces` warning on clang. Add another set of curly braces.

The warnings seems spurious: https://bugs.llvm.org/show_bug.cgi?id=21629
For now this is an easy fix though.

Reply:
Hmmm, that's a very strange warning to get. I've added another set of braces, hopefully that suppresses it.